src/test-apps/TestRetainedPacketBuffer.cpp

```diff
@@ -258,7 +275,7 @@ static void CheckRetainAllocatedBufferWithImplicitRelease(nlTestSuite *inSuite,
 
     // Assert that the buffer has been released to the pool, as expected.
 
-    NL_TEST_ASSERT(inSuite, System::Stats::GetResourcesInUse()[System::Stats::kSystemLayer_NumPacketBufs] == 0);
+    NL_TEST_ASSERT_NO_BUFFERS_IN_USE(inSuite);
```

Comment:
Unable to channel my inner Giuseppe, what are we losing by eliminating the fault injection / stats?

Reply:
Actually all I did was define a custom assert macro (see above) that asserts there are no buffers in use, using the same System::Stats interface.  This made is easier to vary the assertion between testing one stat value (in the case of uniform sized buffers) verses three values (for variable-sized buffers).

Reply:
Missed that, thanks.